pkg/ui/workspaces/cluster-ui/src/api/nodesApi.ts

```diff
@@ -9,12 +9,57 @@
 // licenses/APL.txt.
 
 import { cockroach } from "@cockroachlabs/crdb-protobuf-client";
+import { useMemo } from "react";
+import useSWR from "swr";
 
 import { fetchData } from "src/api";
+import { getRegionFromLocality } from "src/store/nodes";
 
-const NODES_PATH = "_status/nodes";
+import { NodeID, StoreID } from "../types/clusterTypes";
+
+const NODES_PATH = "_status/nodes_ui";
 
 export const getNodes =
   (): Promise<cockroach.server.serverpb.NodesResponse> => {
     return fetchData(cockroach.server.serverpb.NodesResponse, NODES_PATH);
   };
+
+export const useNodeStatuses = () => {
+  const { data, isLoading, error } = useSWR(NODES_PATH, getNodes, {
+    revalidateOnFocus: false,
+  });
+
+  const nodeIDToRegion = useMemo(() => {
```

Comment:
My react is a bit rusty, and I'm not too familiar with SWR, but why do we need to use `useMemo` here?

Reply:
It's so that we only recompute this value when the data changes. This hook will get called on every re-render, but since the data changes less ferquently than that we can useMemo can be used to only recompute items when necessary.

Reply:
For the SWR hook, you mostly just need to know that it caches based on a unique key, which in this case is just the nodes api route path, since we don't provide any params to it (see https://swr.vercel.app/#overview for more if you're curious).
So for now, because we're also not polling this nodes data yet (we can set that in the swr options param), these maps will only be computed once when the data loads, instead of every time the parent component re-renders (which is often).